aspnetcore/grpc/health-checks.md

```diff
@@ -0,0 +1,86 @@
+
+* Add a `Grpc.AspNetCore.HealthChecks` package reference.
+* Register gRPC health checks service in `Startup.cs`:
+  * `AddGrpcHealthChecks` to register services that enable health checks.
+  * `MapGrpcHealthChecksService` to add a health checks service endpoint.
+* Add health checks by implementing <xref:Microsoft.Extensions.Diagnostics.HealthChecks.IHealthCheck> or using the <xref:Microsoft.Extensions.DependencyInjection.HealthChecksBuilderAddCheckExtensions.AddCheck%2A> method.
+
+[!code-csharp[](~/grpc/health-checks/Startup.cs?name=snippet_1&highlight=4-6,16)]
+
+When health checks is set up:
+
+* The health checks service is added to the server app.
+* .NET health checks registered with the app are periodically executed for health results. Health results determine what the gRPC service reports:
+  * `Unknown` is reported when there are no health results.
+  * `NotServing` is reported when there are any health results of <xref:Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus.Unhealthy?displayProperty=nameWithType>.
+  * Otherwise, `Serving` is reported.
+
+## Configure `Grpc.AspNetCore.HealthChecks`
+
+By default, the gRPC health checks service uses all registered health checks to determine health status. gRPC health checks can be customized when registered to use a subset of health checks. The `MapService` method is used to map health results to service names, along with a predicate for filtering health results:
+
+```csharp
+services.AddGrpcHealthChecks(o =>
+{
+    o.Services.MapService("", r => r.Tags.Contains("public"));
+});
+```
+
+The preceding code overrides the default service (`""`) to only use health results with the "public" tag.
+
+gRPC health checks supports the client specifying a service name argument when checking health. Multiple services are supported by providing a service name to `MapService`:
+
+```csharp
+services.AddGrpcHealthChecks(o =>
+{
+    o.Services.MapService("", r => r.Tags.Contains("public"));
+    o.Services.MapService("greet.Greeter", r => r.Tags.Contains("public") && r.Tags.Contains("greeter"));
+    o.Services.MapService("count.Counter", r => r.Tags.Contains("public") && r.Tags.Contains("counter"));
+});
+```
+
+The service name is usually the default (`""`) or a package qualified name of a service in your app. However, nothing prevents using arbitrary values to check app health.
+
+## Call gRPC health checks service
+
+The [`Grpc.HealthCheck`](https://www.nuget.org/packages/Grpc.HealthCheck) package includes a client for gRPC health checks.
+
+```csharp
+var channel = GrpcChannel.ForAddress("https://localhost:5001");
+var client = new Health.HealthClient(channel);
+
+var response = client.CheckAsync(new HealthCheckRequest());
```

Comment:
Could this be 
```suggestion
var response = client.CheckAsync(HealthCheckRequest.Default);
```
to avoid the repeated allocation of the `HealthCheckRequest`?

PS: I see here it's the docs, not the implementation (which I probably missed).

Reply:
A default instance wouldn't make sense as the request object is mutable.